plutus-tx-plugin/test/IsData/9.6/matchAsData.pir-readable.golden

```diff
@@ -0,0 +1,209 @@
+let
+  data (Maybe :: * -> *) a | Maybe_match where
```

Comment:
I guess this (and other datatypes in PIR) will go away once we apply the same approach to `PlutusTx.Maybe`? Is that the plan?

Reply:
I'd be very wary of applying this approach to Maybe, that would make it no longer a generically usable datatype. We could define a separate type for the Data version, I guess. Then we'd really need an entirely different library with copies of almost everything...

Also, since normal datatypes and AsData datatypes can interoperate, it's not necessarily a problem. If you have a field of type `Maybe X` in an AsData type where X is also an AsData type, you just add one level of unwrapping and wrapping when dealing with it, which isn't too bad.